sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt

```diff
@@ -30,46 +26,74 @@ fun TopBarLib() {
 
 @Composable
 fun DisplayScreenSetup(viewModel: SportFeatureViewModelLib = viewModel()) {
-    MainScreen(viewModel.sharedFlow)
+    MainScreen(viewModel)
 }
 
 @Composable
-//Todo: read up on side effects in JetPack Compose with coroutines - would not put this code into production until I understood these ideas better
-fun MainScreen(sharedFlow: SharedFlow<List<Sport>>) {
+fun RefreshButtonLib(viewModel: SportFeatureViewModelLib = viewModel()) {
+    val scope = rememberCoroutineScope()
+    // In practice would pull the last sport displayed from Shared Prefs.
+    var currentSport = Sport.createMockedSports()[0]
 
-    val messages = remember { mutableStateListOf<List<Sport>>() }
+    val getNewSportOnClick: () -> Unit = {
+        scope.launch {
+            try {
 
-    LaunchedEffect(key1 = Unit) {
-        sharedFlow.collect {
-            messages.add(0, it)
-            // Todo: Learn more about recomposition so I can find a better way to achieve this
-            messages.removeRange(1,messages.size)
+                // try and collect the value from the flow and then print it out
```

Comment:
Do we want to add this code here? Pros/Cons?

Reply:
Yes, this sort of processing should be done in the ViewModel. In the improved branch _get_refresh_button_working_ this code doesn't exist. Initially I was using SharedFlow but by using the other Flow API, StateFlow, I could easily access the last value emitted by doing state.value